packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx

```diff
@@ -51,35 +40,45 @@ const LinkContent = React.forwardRef<HTMLAnchorElement, LinkContentProps>(
     const [popoverOpen, setPopoverOpen] = React.useState(
       editor.lastInsertedLinkPath ? Path.equals(path, editor.lastInsertedLinkPath) : false
     );
-    const [isEditing, setIsEditing] = React.useState(link.url === '');
     const linkRef = React.useRef<HTMLAnchorElement>(null!);
     const elementText = link.children.map((child) => child.text).join('');
     const [linkText, setLinkText] = React.useState(elementText);
     const [linkUrl, setLinkUrl] = React.useState(link.url);
+    const linkInputRef = React.useRef<HTMLInputElement>(null);
+    const [showRemoveButton, setShowRemoveButton] = React.useState(false);
+    const [invalidLinkError, setInvalidLinkError] = React.useState('');
 
     const handleOpenEditPopover: React.MouseEventHandler<HTMLAnchorElement> = (e) => {
       e.preventDefault();
       setPopoverOpen(true);
+      setShowRemoveButton(true);
     };
 
     const handleSave: React.FormEventHandler = (e) => {
       e.stopPropagation();
 
-      // If the selection is collapsed, we select the parent node because we want all the link to be replaced)
-      if (editor.selection && Range.isCollapsed(editor.selection)) {
-        const [, parentPath] = Editor.parent(editor, editor.selection.focus?.path);
-        Transforms.select(editor, parentPath);
-      }
-
-      editLink(editor, { url: linkUrl, text: linkText });
-      setIsEditing(false);
-    };
-
-    const handleCancel = () => {
-      setIsEditing(false);
-
-      if (link.url === '') {
-        removeLink(editor);
+      try {
+        // eslint-disable-next-line no-new
+        new URL(linkUrl);
+
+        // If the selection is collapsed, we select the parent node because we want all the link to be replaced)
+        if (editor.selection && Range.isCollapsed(editor.selection)) {
+          const [, parentPath] = Editor.parent(editor, editor.selection.focus?.path);
+          Transforms.select(editor, parentPath);
+        }
+
+        editLink(editor, { url: linkUrl, text: linkText });
+        setPopoverOpen(false);
+        editor.lastInsertedLinkPath = null;
+      } catch (error) {
+        e.preventDefault();
+
+        setInvalidLinkError(
+          formatMessage({
+            id: 'components.Blocks.popover.link.error',
+            defaultMessage: 'Please enter valid link',
+          })
+        );
```

Comment:
The UX chosen in the designs is different, it's to just disable the field if the value is invalid and not show an error message